web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1177,27 +1212,44 @@ namespace com.keyman.keyboards {
                 cmd = cmd + '$';
               }
 
+              languageFound = true;
               break;
             }
           }
+          if (!languageFound) {
+            // TODO: Add error msg that no keyboards are available for lgName
+            console.error('No keyboards are available for '+lgName+'. '+
+              'Does it have another language name?');
+          }
         }
 
         if(cmd == '') {
-          this.keymanweb.util.alert('No keyboards are available for '+languages[0]+'. '
-            +'Does it have another language name?');
+          // TODO: Check useAlert flag to determine whether or not KeymanWeb should display its own alert messages
+          let msg = 'No keyboards are available for '+languages[0]+'. '
+              +'Does it have another language name?';
+          this.keymanweb.util.alert(msg);
+          return Promise.reject(new Error(msg));
         } else {
-          this.keymanCloudRequest('&keyboardid='+cmd,false);
+          return this.keymanCloudRequest('&keyboardid='+cmd, false);
         }
       }
+
+      if (retPromise) {
+        return retPromise;
+      } else {
+        console.log('Should have something here');
```

Comment:
This `console.log()` looks like a development artifact and should not remain.  If there is a reason to leave it, please make the message much more informative for KMW engine users (developers of sites using KMW).